src/main/java/com/gregtechceu/gtceu/api/data/chemical/material/Material.java

```diff
@@ -1060,193 +1243,514 @@ public Builder ignoredTagPrefixes(TagPrefix... prefixes) {
             return this;
         }
 
+        /**
+         * Add a custom Item Tag to all items made from this Material.
+         *
+         * @param key The tag to add.
+         */
+        public Builder customTags(TagKey<Item> key) {
+            this.itemTags.add(key);
+            return this;
+        }
+
+        /**
+         * Set the Element of this Material.<br>
+         * Should be effectively singleton; each element should only have 1 Material claiming to represent it.
+         *
+         * @param element The {@link Element} that this Material represents.
+         */
         public Builder element(Element element) {
             this.materialInfo.element = element;
             return this;
         }
 
+        /**
+         * Set the Formula of this Material.
+         * <br>
+         * <br>
+         * Will override the automatically generated formula.<br>
+         * Will automatically format numbers as subscripts.
+         *
+         * @param formula The formula for this Material.
+         */
         public Builder formula(String formula) {
             this.formula = formula;
             return this;
         }
 
         /**
-         * Replaced the old toolStats methods which took many parameters.
-         * Use {@link ToolProperty.Builder} instead to create a Tool Property.
+         * Set the Formula of this Material.
+         * <br>
+         * <br>
+         * Will override the automatically generated formula.<br>
+         *
+         * @param formula        The formula for this Material.
+         * @param withFormatting Whether numbers should be formatted as subscripts.
+         */
+        public Builder formula(String formula, boolean withFormatting) {
```

Comment:
New function to allow disabling the automatic formatting for formulas in the Builder